pkg/kube/snapshot/snapshot_alpha.go

```diff
@@ -259,14 +261,16 @@ func (sna *SnapshotAlpha) GetSource(ctx context.Context, snapshotName, namespace
 }
 
 // CreateFromSource will create a 'Volumesnapshot' and 'VolumesnaphotContent' pair for the underlying snapshot source.
-func (sna *SnapshotAlpha) CreateFromSource(ctx context.Context, source *Source, snapshotName, namespace string, waitForReady bool, labels map[string]string) error {
+func (sna *SnapshotAlpha) CreateFromSource(ctx context.Context, source *Source, snapshotName, namespace string, waitForReady bool, labels map[string]string, annotations map[string]interface{}) error {
 	deletionPolicy, err := sna.getDeletionPolicyFromClass(source.VolumeSnapshotClassName)
 	if err != nil {
 		return errkit.Wrap(err, "Failed to get DeletionPolicy from VolumeSnapshotClass")
 	}
 	contentName := snapshotName + "-content-" + string(uuid.NewUUID())
-	snap := UnstructuredVolumeSnapshotAlpha(snapshotName, namespace, "", contentName, source.VolumeSnapshotClassName, blockstorage.SanitizeTags(labels))
-	if err := sna.CreateContentFromSource(ctx, source, contentName, snapshotName, namespace, deletionPolicy); err != nil {
+	snapshotAnnotation := getAnnotation(annotations, SnapshotAnnotation)
+	snap := UnstructuredVolumeSnapshotAlpha(snapshotName, namespace, "", contentName, source.VolumeSnapshotClassName, blockstorage.SanitizeTags(labels), snapshotAnnotation)
+	contentAnnotation := getAnnotation(annotations, SnapshotContentAnnotation)
```

Comment:
are we getting volumesnapshot annotations and the volumesnapshotcontent annotations from the same annotations map that is passed to this method?
how are we differentiating between the two?
I think it will become confusing for the callers.

Reply:
Yes, here we are creating map[string]interface{} in the case where string can be `snapshot` and `snapshotcontent` and value is interface which can again be map[string]string for each snapshot and snapshot content annotation.
E.g: 
```
testAnnotation := map[string]interface{}{
		SnapshotAnnotation : map[string]string{
			"testannotation" : "value",
			"testannotation1": "value1",
		},
		SnapshotContentAnnotation : map[string]string{
			"testannotation2" : "value2",
			"testannotation3": "value3",
		},
	}
```